### PR TITLE
serac spack pkg: add +asan variant

### LINKED ISSUE #531: Add `asan` variant to serac package
Downstream customers have asked to be able to toggle our ability to build our code with AddressSanitizer turned on.  Having this turned on in many packages allows better memory error detection. This should be easy in the serac spack package.

### PR DESCRIPTION
**Pull request by goxberry**
This commit adds a `+asan` variant to the Serac Spack package to
expose more readily the `ENABLE_ASAN` option to users building Serac
through Spack and Uberenv. This change is probably more convenient for
downstream users of Serac that build it as a dependency of their
projects via Spack/Uberenv.

Resolves #531.

### CODE REVIEW

File: scripts/spack/packages/serac/package.py [hunk 1]
```diff
@@ -47,6 +47,8 @@ class Serac(CachedCMakePackage, CudaPackage):
             description='Enable runtime safety and debug checks')
     variant('shared',   default=False,
             description='Enable build of shared libraries')
+    variant('asan', default=False,
```
Comment (white238) [inline scripts/spack/packages/serac/package.py:51]: Can you add a conflict if the compiler is not clang or gcc?

Comment (goxberry) [reply scripts/spack/packages/serac/package.py:51]: Good catch! Totally forgot about that restriction.

Comment (goxberry) [reply scripts/spack/packages/serac/package.py:51]: Added checks in [`8916e60f`](https://github.com/LLNL/serac/pull/571/commits/8916e60f5852ee7247e62886bb762fcbebc2d9bc). The checks might be overkill -- comments explain the approach.

**Review by white238 — APPROVED** [review]

**Comment by jamiebramwell** [comment]
I'm going to go investigate why this is failing CI

**Review by jamiebramwell — APPROVED** [review]
Thanks @goxberry ! Sorry for the delay. CI woes. 😭

**Pull request merged**